ipnsfs/system.go

```diff
@@ -0,0 +1,285 @@
+package ipnsfs
```

Comment:
it seems to me like most of this package can be a generalized "MutableFS" or something, and only the top is ipns specific-- i may be missing something